From 768ce710b510205907145fd9f96096ca59e1d3af Mon Sep 17 00:00:00 2001
From: Modestas Vainius <modestas@vainius.eu>
Date: Thu, 19 Nov 2009 20:58:39 +0200
Subject: [PATCH 3/3] cmake build+ steps need Makefile.

The condition is not what dh_auto_* 7.0.x would have done. The
patch makes auto-selection to pass through cmake.pm if Makefile
was not created. This problem is not very dangerous though.
---
 Debian/Debhelper/Buildsystem/cmake.pm |    5 +++--
 t/buildsystems/buildsystem_tests      |    2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/Debian/Debhelper/Buildsystem/cmake.pm b/Debian/Debhelper/Buildsystem/cmake.pm
index ee90c1f..24f486f 100644
--- a/Debian/Debhelper/Buildsystem/cmake.pm
+++ b/Debian/Debhelper/Buildsystem/cmake.pm
@@ -17,12 +17,13 @@ sub check_auto_buildable {
 	my $this=shift;
 	my ($step)=@_;
 	if (-e $this->get_sourcepath("CMakeLists.txt")) {
-		my $ret = $this->SUPER::check_auto_buildable(@_);
+		my $ret = ($step eq "configure" && 1) ||
+		          $this->SUPER::check_auto_buildable(@_);
 		# Existence of CMakeCache.txt indicates cmake has already
 		# been used by a prior build step, so should be used
 		# instead of the parent makefile class.
 		$ret++ if ($ret && -e $this->get_buildpath("CMakeCache.txt"));
-		return $ret > 0 ? $ret : 1;
+		return $ret;
 	}
 	return 0;
 }
diff --git a/t/buildsystems/buildsystem_tests b/t/buildsystems/buildsystem_tests
index e19ab5d..d8619fc 100755
--- a/t/buildsystems/buildsystem_tests
+++ b/t/buildsystems/buildsystem_tests
@@ -256,7 +256,7 @@ touch "$tmpdir/configure", 0755;
 test_check_auto_buildable($bs{autoconf}, "configure", { configure => 1 });
 
 touch "$tmpdir/CMakeLists.txt";
-test_check_auto_buildable($bs{cmake}, "CMakeLists.txt", 1);
+test_check_auto_buildable($bs{cmake}, "CMakeLists.txt", { configure => 1 });
 
 touch "$tmpdir/Makefile.PL";
 test_check_auto_buildable($bs{perl_makemaker}, "Makefile.PL", { configure => 1 });
-- 
1.6.5.2

